test/ecto/query_test.exs

```diff
@@ -268,6 +268,42 @@ defmodule Ecto.QueryTest do
     end
   end
 
+  describe "named joins" do
+    test "assigns a name to a join" do
```

Comment:
Those tests are very fragile because they are asserting on the AST. The reason why we test them above is because we want to check if the query is optimizable but we are not worried about optimization here. My recommendation is to build actual queries and then traverse `query.joins` (or `query.aliases` if we follow my proposal) checking for the generated aliases.

Reply:
It is also very important to test what happens when we give an alias that already exists. We also need to check what happens when you try to use an alias that does not exist.